pkg/api/types.go

```diff
@@ -1980,3 +1980,28 @@ type MetadataWithTest struct {
 func (m *MetadataWithTest) JobName(prefix string) string {
 	return m.Metadata.JobName(prefix, m.Test)
 }
+
+type Architecture string
+
+const (
+	AMD64Arch Architecture = "amd64"
+	ARM64Arch Architecture = "arm64"
+)
+
+var archToCluster = map[Architecture]Cluster{
+	ARM64Arch: ClusterARM01,
+}
+
+func (a Architecture) IsValid() (bool, error) {
+	if _, err := a.GetDefaultCluster(); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (a Architecture) GetDefaultCluster() (Cluster, error) {
+	if c, ok := archToCluster[a]; ok {
+		return c, nil
+	}
+	return "", fmt.Errorf("no predefined cluster for arch: %s", string(a))
```

Comment:
I don't think we should return an error at all. If the cluster is not valid, let's just return false, and perhaps a message with that will include all supported clusters.